jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelState.java

```diff
@@ -819,11 +819,16 @@ protected void onError(Throwable th)
         {
             // If we are still async and nobody has called sendError
             if (_requestState == RequestState.ASYNC && !_sendError)
+            {
                 // Then the listeners did not invoke API methods
                 // and the container must provide a default error dispatch.
                 sendError(th);
-            else
+            }
+            // Otherwise the listeners have called AsyncContext.complete().
+            else if (_requestState != RequestState.COMPLETE)
```

Comment:
Is this an unrelated fix?

Reply:
Correct, it has been introduced in a separate commit, 9dec284cc2afeeb49d3f4fce4a0656d8d98c6d48.